pkg/daemon/node.go

```diff
@@ -49,19 +49,32 @@ func getNodeAnnotation(client corev1.NodeInterface, node string, k string) (stri
 	return getNodeAnnotationExt(client, node, k, false)
 }
 
-// getNodeAnnotationExt is like getNodeAnnotation, but allows one to customize ENOENT handling
-func getNodeAnnotationExt(client corev1.NodeInterface, node string, k string, allowNoent bool) (string, error) {
+// GetNode gets the node object.
+func GetNode(client corev1.NodeInterface, node string) (*core_v1.Node, error) {
+	var lastErr error
 	var n *core_v1.Node
 	err := wait.PollImmediate(10*time.Second, 5*time.Minute, func() (bool, error) {
-		var err error
-		n, err = client.Get(node, metav1.GetOptions{})
-		if err == nil {
+		n, lastErr = client.Get(node, metav1.GetOptions{})
+		if lastErr == nil {
 			return true, nil
 		}
+		glog.Warningf("Failed to fetch node %s (%v); retrying...", node, lastErr)
 		return false, nil
 	})
 	if err != nil {
-		return "", err
+		if err == wait.ErrWaitTimeout {
+			glog.Warningf("Timed out trying to fetch node %s; last error: %v", lastErr)
+		}
+		return nil, lastErr
```

Comment:
so, this is fine, but we can still wrap the time out error https://github.com/openshift/machine-config-operator/blob/master/pkg/operator/sync.go#L397-L400

Reply:
Ahh OK, I thought the goal of #410 was keeping the same error so it could be matched higher up in the stack. If we wrap it, we're losing information, right?

Reply:
As an example, the instance in `updateNodeRetry()` is matching against `errors.IsConflict(err)`.

Reply:
well, mmm yeah, wrapping is something we still need to do for these kinds of error. Example: say this times out for some reason, callers want to know 1) is it a timeout? 2) what was the underlying error? 
The code right now isn't telling callers that we actually timed out right? we need to bubble that up to make logic around permanent/temporarly failures (?)

The example here https://github.com/openshift/machine-config-operator/blob/master/pkg/operator/sync.go#L397-L400 is telling you both 1) && 2) even we should start using `pkg/errors.Wrap` which has a `.Cause()` method which will tell us more about the wrapped error

Reply:
I'm talking about this https://github.com/pkg/errors#adding-context-to-an-error

Reply:
Ahh yes, `Cause()` is exactly what I mean. I didn't realize that's what you meant since we don't currently do that in https://github.com/openshift/machine-config-operator/blob/master/pkg/operator/sync.go#L397-L400. K, will update to use `Wrap`!

Reply:
> Ahh yes, `Cause()` is exactly what I mean. I didn't realize that's what you meant since we don't currently do that in https://github.com/openshift/machine-config-operator/blob/master/pkg/operator/sync.go#L397-L400. K

well, by wrapping in that snippet I meant that we were still _wrapping_ the timeout err with lastErr https://github.com/openshift/machine-config-operator/blob/master/pkg/operator/sync.go#L398